File: backend/app/schemas/shipping.py
from datetime import datetime

from pydantic import BaseModel


class ShipmentCreate(BaseModel):
    order_id: int
    shipping_account_number: str | None = None
    carrier: str | None = None
    tracking_number: str | None = None
    shipped_at: datetime | None = None
    notes: str | None = None


class ShipmentOut(BaseModel):
    id: int
    order_id: int
    shipping_account_number: str | None = None
    carrier: str | None = None
    tracking_number: str | None = None
    shipped_at: datetime | None = None
    notes: str | None = None

    model_config = {"from_attributes": True}

File: frontend/src/api/shipping.ts
import client from "./client";

export interface Shipment {
  id: number;
  order_id: number;
  shipping_account_number?: string;
  carrier?: string;
  tracking_number?: string;
  shipped_at?: string;
  notes?: string;
}

export const shippingApi = {
  list: () => client.get<Shipment[]>("/shipments").then((r) => r.data),
  get: (id: number) => client.get<Shipment>(`/shipments/${id}`).then((r) => r.data),
  create: (data: Partial<Shipment>, turnstileToken?: string) =>
    client
      .post<Shipment>("/shipments", data, turnstileToken ? { headers: { "CF-Turnstile-Response": turnstileToken } } : {})
      .then((r) => r.data),
};
